imagebuildah/stage_executor.go

```diff
@@ -1165,6 +1185,15 @@ func (s *StageExecutor) Execute(ctx context.Context, base string) (imgID string,
 			commitName = s.output
 		}
 
+		// If --cache-from or --cache-to is specified make sure to populate
+		// cacheKey since it will be used either while pulling or pushing the
+		// cache images.
+		if s.executor.cacheFrom != nil || s.executor.cacheTo != nil {
```

Comment:
Non-blocking: I wouldn’t mind computing this always. The other two updates of `cacheKey` are unconditional, and that inconsistency seems unnecessary and potentially confusing to future readers.

The cost of `generateCacheKey` is not zero, but it’s likely trivial compared to the build steps or cache use.

Alternatively, `needsCacheKey := s.executor.cacheFrom != nil || s.executor.cacheTo != nil` and wrap _all_ updates of `cacheKey` with `if needsCacheKey`.